### PR TITLE
LPS-133354 Mime Types filter

### PR DESCRIPTION
**Pull request by 4lejandrito**
Motivation
=======
[Link to story](https://issues.liferay.com/browse/LPS-133354)

Proposed Solution
========
Developed new feature for filtering content by file extension in Content Dashboard.

Change summary:
---------------
* TODO

Steps to Verify:
----------------
1. Add some documents and images to site: Site menu -> Content & Data -> Documents and media
2. Go to content dashboard and start filtering the data in the table by file extension and check that everything works as expected.

Screenshots (optional):
-----------------------
Captura de pantalla 2021-08-18 a las 16 30 54
Captura de pantalla 2021-08-18 a las 16 31 24
Captura de pantalla 2021-08-18 a las 16 31 48

### CODE REVIEW

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:stop

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:forward

**Review by YurenaC — APPROVED** [review]
Backend changes approved after review and discussion with @4lejandrito

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectFileExtension.js [hunk 1]
```diff
@@ -42,143 +41,18 @@ const nodeTreeArrayMapper = (nodeArray) => {
...
-								</div>
-							</div>
-						)}
-					</div>
-				</ClayLayout.ContainerFluid>
-			</form>
-		</div>
+		<TreeFilter
+			itemSelectorSaveEvent={itemSelectorSaveEvent}
+			necessaryKeys={['id']}
```
Comment (beltranrengifo) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectFileExtension.js:52]: `necessaryKeys` I don't like very much this name 
Maybe can be use 'mandatory' instead on 'necessary' and add something about the purpose?
Something like `mandatoryBackendKeys`, `backendMandatoryProperties` ... I'm not sure

**Review by beltranrengifo — COMMENTED** [review]
Hi Clara! Nice work with this PR and the Treefilter component 🎉  !!
Just a little comment, let me know your thoughts

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectFileExtension.js (hunk 1 above)
Comment (claraizquierdo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectFileExtension.js:52]: Yeah, I struggled finding an appropriate name for that prop 🤣

**Review by beltranrengifo — APPROVED** [review]
👍

**Comment by claraizquierdo** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
ci:reevaluate:1307401_383

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:stop

**Comment by 4lejandrito** [comment]
ci:forward

**Pull request closed without merging**